include/libdnf/module/module_item_container.hpp

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_ITEM_CONTAINER_HPP
+#define LIBDNF_MODULE_MODULE_ITEM_CONTAINER_HPP
+
+#include "libdnf/base/base_weak.hpp"
+#include "libdnf/common/weak_ptr.hpp"
+#include "libdnf/module/module_item.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace libdnf::module {
+
+
+class ModuleItemContainer;
+
+
+using ModuleItemContainerWeakPtr = libdnf::WeakPtr<ModuleItemContainer, false>;
+
+
+class ModuleItemContainer {
+public:
+    enum class ModuleState { UNKNOWN, ENABLED, DISABLED, DEFAULT, INSTALLED };
```

Comment:
Just note. May be UNKNOWN is not the best state but I don't see a better option.